idaes/surrogate/plotting/sm_plotter.py

```diff
@@ -60,64 +68,59 @@ def scatter2D(xdata, zdata, xtest, zfit, xlabels=None, zlabels=None, dz=None,
         zlabels = []
         for j in range(numouts):
             zlabels.append('z' + str(j+1))
-    if clabel is None:
-        clabel = 'Confidence Intervals'
 
-    pointidx = -1  # dummy index for each plot produced
-    fig, ax = [], []
+    if filename is not None:
+        if filename[-4:] != '.pdf':  # adding extension if not present
```

Comment:
We might want to be careful here. This will just append `.pdf` to the filename if the last four characters are not `.pdf`. However, if the user specified an extension that was not `.pdf` we would get something like `foo.bar.pdf` as the filename which might cause some problems (depending on OS).

Reply:
Testing some specific cases, saving to '.xps' and no extension produce files that show the plot images with no issues (the second one is strange to me), and even saving to '.txt' produces a text file successfully. I initially wanted to ensure the PdfPages() call would print to .pdf, although we could just let the user enter any name including the extension they want.

Reply:
In many ways, the extension itself doesn't matter - it merely tells the OS what programme to use to open the file by default. The more important issue however is how the OS would handle a case where there are two periods in the file name. I think a lot of OS's can handle that these days (not that I try), but it used to be a problem and should probably be avoided. There is also a further issue that the user might not be aware that we append `.pdf` to the file name if it is not present, and might deliberately use the name `foo.bar` and expect the resulting file to have that name.

For sanity, I would often take the most restrictive approach and instead raise and `Exception` if the filename does not end in `.pdf`. That way there is no ambiguity.

Reply:
I did try '[name].bar.pdf' and it opened normally on my Windows machine, however I agree with you about enforcing '.pdf' in the filename.